EIPS/eip-3978.md

```diff
@@ -0,0 +1,52 @@
+---
+eip: 3978
+title: Gas refunds on reverts
+author: Anton Bukov (@k06a), Mikhail Melnik (@ZumZoom)
+discussions-to: <to be assigned>
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-09-16
+---
+
+## Simple Summary
+Do not zeroify gas refunds of transaction subcall reverts, due users pay a lot of gas for storage non-modification.
+    
+## Abstract
+>A short (~200 word) description of the technical issue being addressed. This should be a very terse and human-readable version of the specification section. Someone should be able to read only the abstract to get the gist of what this specification does.
+
+Since [EIP-3298](https://eips.ethereum.org/EIPS/eip-3298) gas refunds works for storage restores only inside the same transaction. For example ERC20 `approve` + `transferFrom` flow between 2 smart contracts according to [EIP-2200](https://eips.ethereum.org/EIPS/eip-2200) and [EIP-2929](https://eips.ethereum.org/EIPS/eip-2929) will cost nearly to `21600` gas with gas refund counter `20000`. But in case of reverting this subcall (containing both `approve` and `transferForm`) gas refund will be erased, while smart contract storage will remain unmodified. I think it should keep storage access costs, but still refund modification costs.
```

Comment:
```suggestion
Since [EIP-3298](https://eips.ethereum.org/EIPS/eip-3298) gas refunds works for storage restores only inside the same transaction. For example [ERC-20](./eip-20.md) `approve` + `transferFrom` flow between 2 smart contracts according to [EIP-2200](https://eips.ethereum.org/EIPS/eip-2200) and [EIP-2929](https://eips.ethereum.org/EIPS/eip-2929) will cost nearly to `21600` gas with gas refund counter `20000`. But in case of reverting this subcall (containing both `approve` and `transferForm`) gas refund will be erased, while smart contract storage will remain unmodified. I think it should keep storage access costs, but still refund modification costs.
```